### ISSUE TITLE
Setting bootorder via EFI vars sometimes doesn't work

### ISSUE THREAD

**Issue by marmarek** [Dasharo/dasharo-issues#303]
**Dasharo version**

v1.1.0

**Dasharo variant**

MSI

**Affected component(s) or functionality**

**Brief summary**

Setting boot order via `efibootmgr` sometimes fails with `Invalid argument`. Retrying the same value works (sometimes after 2-3 tries).

**How reproducible**

Often, but not 100% time.

**How to reproduce**

1. `efibootmgr` - note boot order
2. `efibootmgr -o ...` - set new boot order (different than it was before)

**Expected behavior**

Works each time.

**Actual behavior**

Sometimes fails like this:
```
[root@dom0 anaconda]# efibootmgr 
BootCurrent: 0004
Timeout: 3 seconds
BootOrder: 0004,0003,0002,0000
Boot0000* Setup
Boot0001  UEFI BootManagerMenuApp
Boot0002* iPXE Network Boot
Boot0003* UEFI Shell
Boot0004* Qubes OS
[root@dom0 anaconda]# efibootmgr -o 0002,0000
Could not set BootOrder: Invalid argument
[root@dom0 anaconda]# efibootmgr -o 0002,0000
BootCurrent: 0004
Timeout: 3 seconds
BootOrder: 0002,0000
Boot0000* Setup
Boot0001  UEFI BootManagerMenuApp
Boot0002* iPXE Network Boot
Boot0003* UEFI Shell
Boot0004* Qubes OS
```

**Screenshots**

**Additional context**

**Solutions you've tried**

**Comment by miczyg1** [comment]
@marmarek efibootmgr invoked from Dom0?

**Comment by marmarek** [comment]
Yes

**Comment by miczyg1** [comment]
Hmm flipped around the first two boot entries according to the bootorder variable for over 10 times and it worked without issues all times. This was on ubuntu 22.04 however. What output you get from `efibootmgr -V`? Mine is `version 17` whatever it means...

**Comment by marmarek** [comment]
`version 16`

For me, this happens at least just after installing the system. In that case, it's like in the above output - removing two entries from the boot order.

**Comment by marmarek** [comment]
I did tried `strace` on it before, and it was indeed EINVAL returned on sysfs write to the variable. Runtime services are proxied via Xen under Qubes, so maybe it is some different handling in native Linux vs Xen? But I do not observe this issue on other (non-Dasharo) systems.

**Comment by miczyg1** [comment]
I will try Qubes as well to reproduce. I assume R4.1.1?

Right, the Xen RS proxy may also be unhappy about something here...

**Comment by marmarek** [comment]
This particular issue was on R4.2, but I think it happens on R4.1 too (unfortunately my CI logs in such case mention only `Boot timeout`, which may be also some other issue).

BTW, I also set BootNext often, and this one seems to work fine.

**Comment by miczyg1** [comment]
@marmarek Ubuntu 22.04 failed at 71th iteration

**Comment by miczyg1** [comment]
@marmarek do we have any updates on this? Did v1.1.1-rc3 help to some extent?

**Comment by marmarek** [comment]
Yes, it did, seems to be okay now. I think I said it on Matrix already, but maybe not...

**Comment by miczyg1** [comment]
@marmarek I also recall such message, but wanted to make sure. Thank yo for confirmation. I assume this issue can be closed when v1.1.1. release is out.
